plugin/src/init.server.lua

```diff
@@ -1,222 +1,289 @@
 local ConnectAction = plugin:CreatePluginAction(
 	"Luau Language Server Connect",
 	"Connect",
 	"Connects to Luau Language Server",
 	"rbxassetid://11115506617",
 	true
 )
+local ConnectButton = Toolbar:CreateButton(
+	"Luau Language Server Setup",
+	"Toggle Menu",
+	"rbxassetid://11115506617",
+	"Luau Language Server"
+)
+local SettingsButton = Toolbar:CreateButton("Settings", "Open Settings", "rbxassetid://13997395868")
 
-local connected = Instance.new("BoolValue")
-local connections = {}
+local DefaultSettingsModule = Instance.new("ModuleScript")
+DefaultSettingsModule.Name = SETTINGS_MODULE_NAME
+ScriptEditorService:UpdateSourceAsync(DefaultSettingsModule, function()
+	return DEFAULT_SOURCE
+end)
+local DefaultSettings: {
+	Port: number,
+	StartAutomatically: boolean,
+	DebugMode: boolean,
+	Include: boolean,
+} =
+	require(DefaultSettingsModule) :: any
+--// Was just much easier to create a permanent reference for the default settings config
+
+local function GetAndValidateSettingsModule()
+	local SettingsModule = (
+		TestService:FindFirstChild(SETTINGS_MODULE_NAME) or Instance.new("ModuleScript")
+	) :: ModuleScript
+	local SettingsModuleSource = ScriptEditorService:GetEditorSource(SettingsModule)
+	if SettingsModuleSource == "" or SettingsModuleSource == nil then
+		SettingsModule = DefaultSettingsModule:Clone()
+		warn(
+			"[Luau Language Server] Could not load settings: Unable to find saved settings, reverting to default settings."
+		)
+	end
 
-type EncodedInstance = {
-	Name: string,
-	ClassName: string,
-	Children: { EncodedInstance },
-}
+	--// No loss in not running a check here since Luau runs an internal validation for set
+	SettingsModule.Name = SETTINGS_MODULE_NAME
+	SettingsModule.Parent = TestService
+
+	local Settings = require(SettingsModule) :: any
+	print(Settings)
+	if typeof(Settings) ~= "table" then
+		warn(
+			"[Luau Language Server] Could not load settings: Settings module does not return a table. Defaulting to default settings."
+		)
+		SettingsModule = DefaultSettingsModule:Clone()
+	elseif typeof(Settings.Port) ~= "number" then
+		warn(
+			`[Luau Language Server] Could not load settings: Port is not a number. Defaulting to {DefaultSettings.Port}.`
+		)
+		Settings.Port = DefaultSettings.Port
+	elseif type(Settings.StartAutomatically) ~= "boolean" then
+		warn(
+			`[Luau Language Server] Could not load settings: StartAutomatically is not a boolean. Defaulting to {DefaultSettings.Port}.`
+		)
+		Settings.StartAutomatically = DefaultSettings.StartAutomatically
+	elseif type(Settings.Include) ~= "table" then
+		warn(
+			`[Luau Language Server] Could not load settings: Include list is not a table. Defaulting to {DefaultSettings.Include}}.`
+		)
+		Settings.Include = DefaultSettings.Include
+	end
 
-local wasConnected
-SettingsButton.Click:Connect(function()
-	plugin:OpenScript(SettingsModule)
-	wasConnected = connected.Value
-end)
+	return SettingsModule
```

Comment:
We perform edits to `Settings` but end up just returning `SettingsModule`, won't this discard all the edits we made?